src/js/components/ConnectionModals/ConnectionForm.tsx

```diff
@@ -95,15 +95,14 @@ const ConnectionForm = ({onClose, conn}: Props) => {
     }
   }
 
-  const toCluster = ({host, name}): Cluster => {
+  const toCluster = ({id, host, name}): Cluster => {
     // set defaults
     let [h, p] = host.split(":")
     if (!p) p = "9867"
-    const hostPort = `${h}:${p}`
     return {
       host: h,
       port: p,
-      id: hostPort,
+      id: id || brim.randomHash(),
```

Comment:
using the `host:port` as the id here, as well as not using an existing id if it already exists was causing a handful of weird bugs where either ids were changing, or you might inadvertently edit a different connection if it's id matched the host:port of the current one you are creating or editing. unique ids make more sense for this anyway.

one note: to help distinguish the "default" connection/workspace we will keep the id for that as "localhost:9867", which both helps us pluck it out of the clusters object more readily and also leaving it like that saves us a migration